microcosm_pubsub/result.py

```diff
@@ -153,6 +153,18 @@ def log(self, logger, opaque):
             },
         )
 
+    def error_reporting(self, sentry_enabled, opaque):
+        opaque = opaque.as_dict()
+        if not sentry_enabled:
+            return
+        from sentry_sdk import capture_exception
+        from sentry_sdk import configure_scope
```

Comment:
If these hadn't been installed, but sentry had been enabled, would you want something friendlier than just raising a bare `ImportError`?

Reply:
The package would be installed it's not optional in the `setup.py`. So we wouldn't have an `ImportError`, the other error that we could have is a missing dsn but that is protected against as setup only succeeds if both enabled flag and dsn is set. Will add some extra handling so if a bad url is setup the sentry is just disabled too.

Reply:
So, we tend to frown a bit on introducing new 3rd-party libs if they aren't central to the functioning of a given lib (idea being that this is open-source, and don't want to push the dep on all downstream consumers). (Yes, I know this isn't widely-used. But it's beside the point).

Also, why not put this as a top-level import if it should be available globally?

Reply:
Fair criticism, I will look to action this if internal POC goes well and pull this out into a separate lib and make it optional within this package.